File: Module_01_Big_O/Topic_02_Constant_Time_O(1)/Python/example_04.py
"""
Course title: AlgoMania
Instructor: Thiago Brito - Universo DEV
Example adapted by: Charles Fernandes de Souza
Date: September 26, 2024
"""
import queue
def example_04():
    my_queue = queue.Queue()
    my_queue.put(1)
    my_queue.put(2)
    my_queue.put(3)
    # The size() method is O(1) (constant time) because it returns
    # the number of elements in the queue without needing to
    # traverse the entire data structure.
    print("Size of the queue:", my_queue.qsize())
example_04()

File: Module_01_Big_O/Topic_02_Constant_Time_O(1)/Java/Example04/src/com/souza/charles/Example04.java
package com.souza.charles;
/**
 Course title: AlgoMania
 Instructor: Thiago Brito - Universo DEV
 Example adapted by: Charles Fernandes de Souza
 Date: September 26, 2024
 */
import java.util.Queue;
import java.util.LinkedList;
public class Example04 {

    public static void main(String[] args) {
        Queue<Integer> myQueue = new LinkedList<>();
        myQueue.add(1);
        myQueue.add(2);
        myQueue.add(3);
        myQueue.add(4);
        // The size() method is O(1) (constant time) because it returns
        // the number of elements in the queue without needing to
        // traverse the entire data structure.
        System.out.println("Size of the queue: " + myQueue.size());
    }
}